model/versioning/Versioned.php

```diff
@@ -780,6 +805,95 @@ public function onAfterSkippedWrite() {
 	}
 
 	/**
+	 * Find all objects owned by the current object.
+	 * Note that objects will only be searched in the same stage as the given record.
+	 *
+	 * @param bool $recursive True if recursive
+	 * @param ArrayList $list Optional list to add items to
+	 * @return ArrayList list of objects
+	 */
+	public function findOwned($recursive = true, $list = null)
+	{
+		// Find objects in these relationships
+		return $this->findRelatedObjects('owns', $recursive, $list);
+	}
+
+	/**
+	 * Find objects which own this object.
+	 * Note that objects will only be searched in the same stage as the given record.
+	 *
+	 * @param bool $recursive True if recursive
+	 * @param ArrayList $list Optional list to add items to
+	 * @return ArrayList list of objects
+	 */
+	public function findOwners($recursive = true, $list = null)
+	{
+		// Find objects in these relationships
+		return $this->findRelatedObjects('owned_by', $recursive, $list);
+	}
+
+	/**
+	 * Find objects in the given relationships, merging them into the given list
+	 *
+	 * @param array $source Config property to extract relationships from
+	 * @param bool $recursive True if recursive
+	 * @param ArrayList $list Optional list to add items to
+	 * @return ArrayList The list
+	 */
+	public function findRelatedObjects($source, $recursive = true, $list = null)
+	{
+		if (!$list) {
+			$list = new ArrayList();
+		}
+
+		// Skip search for unsaved records
+		if(!$this->owner->isInDB()) {
+			return $list;
+		}
+
+		$relationships = $this->owner->config()->{$source};
+		foreach($relationships as $relationship) {
+			// Warn if invalid config
+			if(!$this->owner->hasMethod($relationship)) {
```

Comment:
Does that mean you can declare ownership on arbitrary getters? If it's not the case, we should also check if `$relationship` is actually a DO relationship